internal/scanner/walk.go

```diff
@@ -166,6 +169,16 @@ func (w *Walker) walkAndHashFiles(fchan chan protocol.FileInfo) filepath.WalkFun
 			return skip
 		}
 
+		if w.Pruner != nil {
```

Comment:
If the above simplification is possible, and with a `if p == nil { return false }` in the `func (p*Pruner) ShouldSkip()`, this could all just be a simple if, without nil checks etc...